python_modules/libraries/dagster-sigma/dagster_sigma/resource.py

```diff
@@ -86,197 +88,257 @@ def api_token(self) -> str:
 
         return self._api_token
 
-    def _fetch_json(
+    async def _fetch_json_async(
         self, endpoint: str, method: str = "GET", query_params: Optional[Dict[str, Any]] = None
     ) -> Dict[str, Any]:
         url = f"{self.base_url}/v2/{endpoint}"
         if query_params:
             url = f"{url}?{urllib.parse.urlencode(query_params)}"
 
-        response = requests.request(
-            method=method,
-            url=f"{self.base_url}/v2/{endpoint}",
-            headers={
-                "Accept": "application/json",
-                "Authorization": f"Bearer {self.api_token}",
-                **SIGMA_PARTNER_ID_TAG,
-            },
-        )
-        response.raise_for_status()
-        return response.json()
+        async with aiohttp.ClientSession() as session:
+            async with session.request(
+                method=method,
+                url=f"{self.base_url}/v2/{endpoint}",
+                headers={
+                    "Accept": "application/json",
+                    "Authorization": f"Bearer {self.api_token}",
+                    **SIGMA_PARTNER_ID_TAG,
+                },
+            ) as response:
+                response.raise_for_status()
+                return await response.json()
 
     @cached_method
-    def _fetch_workbooks(self) -> List[Dict[str, Any]]:
-        return self._fetch_json("workbooks")["entries"]
+    async def _fetch_workbooks(self) -> List[Dict[str, Any]]:
```

Comment:
I think using cached methods with async might cause some problems, but you're probably avoiding them by returning the awaited thing so it's probably fine -- do any codepaths currently invoke any of these methods more than once?

Reply:
https://github.com/dagster-io/dagster/pull/25371 😄